service/controller/resource/certificates/resource.go

```diff
@@ -103,16 +102,25 @@ func (r *Resource) getDesiredObject(v interface{}) (*corev1.Secret, error) {
 	secretData := sourceSecret.Data
 
 	if key.IsCAPICluster(cluster) {
-		// CAPI Secret is a kubeconfig so we need to extract the certs from it
+		// CAPI Secret is a kubeconfig so we need to extract the certs or tokens from it
 		if kubeconfig, ok := secretData["value"]; ok {
 			capiKubeconfig, err := clientcmd.Load(kubeconfig)
 			if err != nil {
 				return nil, microerror.Mask(err)
 			}
 			kubeconfigAdminUser := fmt.Sprintf("%s-admin", cluster.GetName())
+			kubeconfigCapiAdminUser := fmt.Sprintf("%s-capi-admin", cluster.GetName())
 			secretData["ca"] = capiKubeconfig.Clusters[cluster.GetName()].CertificateAuthorityData
-			secretData["crt"] = capiKubeconfig.AuthInfos[kubeconfigAdminUser].ClientCertificateData
-			secretData["key"] = capiKubeconfig.AuthInfos[kubeconfigAdminUser].ClientKeyData
+			if capiKubeconfig.AuthInfos[kubeconfigAdminUser] != nil {
+				secretData["crt"] = capiKubeconfig.AuthInfos[kubeconfigAdminUser].ClientCertificateData
+				secretData["key"] = capiKubeconfig.AuthInfos[kubeconfigAdminUser].ClientKeyData
+				secretData["token"] = []byte(capiKubeconfig.AuthInfos[kubeconfigAdminUser].Token)
```

Comment:
I don't get why we are getting a certificate AND token in both conditions. Shouldn't  this be certificate in one and token in the other?

Reply:
Fixed :)